docs/DevGuide/PythonBindings.md

```diff
@@ -35,27 +36,37 @@ function:
 
 \code
 spectre_python_add_module(
-  ExtraDataStructures
+  Extra
+  LIBRARY_NAME "PyExtraDataStructures"
   MODULE_PATH "DataStructures/"
   SOURCES Bindings.cpp MyCoolDataStructure.cpp
   PYTHON_FILES CoolPythonDataStructure.py
   )
 \endcode
 
-The library that is added has the name `PyExtraDataStructures` and requires that
-at least the `${SPECTRE_LINK_PYBINDINGS}` be set as the last entry to the
-`target_link_libraries` call. For example,
+The library that is added has the name `PyExtraDataStructures`. Make sure to
+call `spectre_python_link_libraries` for every Python module that compiles
+`SOURCES`. For example,
 
 \code
-target_link_libraries(
+spectre_python_link_libraries(
   PyExtraDataStructures
   PUBLIC ExtraDataStructures
-  ${SPECTRE_LINK_PYBINDINGS}
   )
 \endcode
 
-\warning `${SPECTRE_LINK_PYBINDINGS}` must be the *last* argument to
-`target_link_libraries`.
+You may also call `spectre_python_add_dependencies` for Python modules that
+have `SOURCES`, e.g.
+
+\code
```

Comment:
I would prefer if these code blocks were changed into actual snippets

Reply:
There's no code in `tests/` that uses these functions, and I think we don't reference code in `src/` in snippets.

Reply:
For C++ or python code we can always put snippets in `tests/` but I've already come across incorrect CMake `\code` blocks (which I'm fixing in a forthcoming PR as a snippet).  I think making sure the documentation is correct is more important than restricting snippets to `tests/`

Reply:
I agree, I think we should move to more snippets, even if that means taking snippets out of the main code base (though from tests is preferred whenever possible)

Reply:
Fine with me. I converted one of the `\code` blocks to a snippet into `tests/` (please check the fixup). The others require changing the `EXAMPLE_PATH` for Doxygen, so I think they deserve a separate PR.

Reply:
#1816